### One Tailed (Right Tailed) - One-Sample Z-Test

**1 - What is it?**

A. `Right One-Tailed One-Sample Z-Test` check the means of sample is significantly higher from the population mean (when population standard devidation is known and sample size is large, usually n>=30).

`Think of i like` is my small group's of average weight `really higher(significantly)` from the entire country's average weight, or it just random chance.

**2- Real-Life Example : Right Tailed One-Sample Z-Test**

`Scenario:` Your marketing teams says: `Out new landing page increase average session time beyond 5 minutes`

So we wanted to test if its really greater than 5 minutes, not just difference - So it's one-tailed test (specifically, right-tailed)

**3 - Hypotheses**

- H0 (Null Hypothesis) : Population mean (meau) = 5 (no improvement).
- H1 (Alternate Hypothesis) : Sample mean > Population mean (mean time improved)

This is a `right-tailed` test because we're checking the `greater than`.

**4 - Formula**

Z = (Sample mean - Population mean) / (Population std/Sqrt(Sample Size))

Z = (x bar - μ) / σ/sqrt(n)

**Parameters:**
- Population mean (μ) : 5 min
- Population std (σ) : 1 min
- Sample mean (x bar) : 5.5 min
- Sample size (n) : 50
- Significance level (alpha) : 0.05
- confidence Interval (CI) : 0.95

Here `population std is know and sample size>=30` so we can do z-test, specifically *right one-tailed one-sample z-test*.

**5 - Python Implementation (Step-by_Step)**

In [5]:
from platform import java_ver
import numpy as np
from scipy import stats

# Step 1 : Define Parameters
population_mean = 5 # average time on site (in minutes)
population_std = 1   # population standard deviation (spread from mean)
sample_mean = 5.5   # sample mean
sample_size = 50    # sample size
alpha = 0.05        # significance level (a)

# Step 2 : Calculate Standard Error (SE)
standard_error = population_std / np.sqrt(sample_size)
print(f"Standard Error: {round(standard_error,4)}")

# Step 3 : Z Statistics
z_stat = (sample_mean - population_mean)/standard_error
print(f"Z-Statistics: {round(z_stat,4)}")

# Step 4 : Calculate Right One-tailed One-sample Z-test P-Value
p_value = 1 - stats.norm.cdf(z_stat)
print(f"P-Value: {round(p_value,4)}")

# Step 5 : Calculate Critical Value (Lower Bound of Critical Region)
# For right-tailed test -> critical value = z_(1-alpha)
critical_value = stats.norm.ppf(1-alpha)
print(f"Critical Value (Right-Tail, a={alpha}): {round(critical_value,4)}")

# Step 6.1 : Decison based on p_value with alpha
if p_value < alpha:
    print("Reject H0: Landing page signficantly increased session time.")
else:
    print("Fail to Reject H0: No significant increase in session time.")

# OR

# Step 6.2 : Decision based on z_stat with critical_value
if z_stat > critical_value:
    print("Reject H0: Landing page significantly increased session time.")
else:
    print("Fail to Reject H0: No significant increase in session time.")

Standard Error: 0.1414
Z-Statistics: 3.5355
P-Value: 0.0002
Critical Value (Right-Tail, a=0.05): 1.6449
Reject H0: Landing page signficantly increased session time.
Reject H0: Landing page significantly increased session time.


**6 - Business Interpretation**

- Z = 3.53 -> Sample mean is 3.5 standard errors above the population mean.
- p = 0.0002 -> There's only 0.02% chance that this result happened by luck.
- So here new page `really did` increase time - on by random luck.